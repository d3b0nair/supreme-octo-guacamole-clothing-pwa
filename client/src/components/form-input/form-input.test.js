import React from "react";
import { shallow } from "enzyme";
import FormInput from "./form-input.component";

let wrapper;
let mockHandleChange;
beforeEach(() => {
  mockHandleChange = jest.fn();
  const mockProps = {
    handleChange: mockHandleChange,
    label: "email",
    value: "test@gmail.com",
  };
  wrapper = shallow(<FormInput {...mockProps} />);
});

describe("FormInput component", () => {
  it("should mount", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render FormInputLabel if there is a label", () => {
    expect(wrapper.exists("FormInputLabel")).toBe(true);
  });
  it("should call handleChange method when input changes", () => {
    wrapper.find("FormInputContainer").simulate("change");
    expect(mockHandleChange).toHaveBeenCalled();
  });
  it("should not render FormInputLabel if there is no label", () => {
    const mockNewProps = {
      label: "",
      value: "test@gmail.com",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInput {...mockNewProps} />);

    expect(newWrapper.exists("FormInputLabel")).toBe(false);
  });
});
