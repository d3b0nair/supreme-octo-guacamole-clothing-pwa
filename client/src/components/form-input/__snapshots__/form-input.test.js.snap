// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should mount 1`] = `
<styled.div>
  <FormInputContainer
    data-testid="FormInputContainer"
    onChange={[MockFunction]}
  />
  <FormInputLabel
    className="shrink form-input-label"
    data-testid="FormInputLabel"
  >
    email
  </FormInputLabel>
</styled.div>
`;
